runironic-api.sh

```diff
@@ -4,4 +4,14 @@ export IRONIC_DEPLOYMENT="API"
 
 . /bin/configure-ironic.sh
 
+# Wait for ironic to load all expected drivers
+# the DB query returns the number of unique hardware_type in the conductor_hardware_interfaces table
+CONF_DRIVERS=$(crudini --get /etc/ironic/ironic.conf DEFAULT enabled_hardware_types | tr ',' '\n' | wc -l)
```

Comment:
I don't know enough about how the plugins work in Ironic, but looking solely at the count makes me a little nervous.

Are there any default or builtin drivers that are always loaded, even if they are not listed in the config?

Would it make sense to compare the list of names, instead of just the count?

We've put some logic in the baremetal-operator to do a similar check, but it just waits for a non-zero count. Should that be made more sophisticated, too? That won't help the OpenShift installer, which the approach here does take care of.

Reply:
> I don't know enough about how the plugins work in Ironic, but looking solely at the count makes me a little nervous.
> 
> Are there any default or builtin drivers that are always loaded, even if they are not listed in the config?

I don't think so, there is a default value for enabled_hardware_types if not mentioned in the config

> Would it make sense to compare the list of names, instead of just the count?

I guess it would be more definite, although I'm not sure it would be worth the extra complexity
 
> 
> We've put some logic in the baremetal-operator to do a similar check, but it just waits for a non-zero count. Should that be made more sophisticated, too? That won't help the OpenShift installer, which the approach here does take care of.

Checking for a non-zero count is not reliable as the drivers are loaded individually, resulting in 
https://bugzilla.redhat.com/show_bug.cgi?id=1902653

Reply:
> Should that be made more sophisticated, too? 

To do this the BMO would need to know what the full list of drivers is

Reply:
In the BMO we could look for the driver we're going to need for the host we're reconciling, so we don't need the full list. @andfasano did that work, I think, so maybe he has ideas about how to extend it to handle this case.

Reply:
Not sure if there's a better place, but we could extract the driver name from the `BareMetalHost.Spec.BMC.Address` of the provisioned host and then check for it in the dependencies check here https://github.com/metal3-io/baremetal-operator/blob/d6eaf6774de775d83bdd47e04ea3d171ba7d4346/pkg/provisioner/ironic/dependencies.go#L66